src/scheduler/fifo.c

```diff
@@ -2407,25 +2408,25 @@ sched_settings_frm_svr(struct batch_status *status)
 		int priv_dir_update_fail = 0;
 		struct attropl *patt;
 		char comment[MAX_LOG_SIZE] = {0};
-		int validation_failed = 0;
-		static char *priv_dir = NULL;
-		static char *log_dir = NULL;
-		if ((log_dir != NULL) && strcmp(log_dir, tmp_log_dir) != 0) {
+
+		if (tmp_log_dir != NULL) {
 			(void)snprintf(path_log,  sizeof(path_log), tmp_log_dir);
 			log_close(1);
 			if (log_open(logfile, path_log) == -1) {
 				/* update the sched comment attribute with the reason for failure */
 				attribs = calloc(2, sizeof(struct attropl));
 				if (attribs == NULL) {
 					schdlog(PBSEVENT_ERROR, PBS_EVENTCLASS_SCHED, LOG_ERR, __func__, MEM_ERR_MSG);
-					free(tmp_log_dir);
-					free(tmp_priv_dir);
-					free(tmp_partitions);
+					if (tmp_log_dir != NULL)
+						free(tmp_log_dir);
+					if (tmp_priv_dir != NULL)
+						free(tmp_priv_dir);
+					if (tmp_partitions != NULL)
+						free(tmp_partitions);
```

Comment:
No need to check against NULL. free(NULL) is a noop.

Reply:
Done